stripe/src/main/java/com/stripe/android/view/CardInputView.java

```diff
@@ -40,9 +55,118 @@ private void initView() {
         inflate(getContext(), R.layout.card_input_view, this);
 
         mScrollView = (LockableHorizontalScrollView) findViewById(R.id.root_scroll_view);
-        mCardNumberEditText = (EditText) findViewById(R.id.et_card_number);
+        mCardNumberEditText = (CardNumberEditText) findViewById(R.id.et_card_number);
         mExpiryDateEditText = (EditText) findViewById(R.id.et_expiry_date);
         mCvcNumberEditText = (EditText) findViewById(R.id.et_cvc_number);
-        mCardNumberSpace = (Space) findViewById(R.id.space_in_container);
+        mCardNumberSpace = findViewById(R.id.space_in_container);
+        mCardNumberIsViewed = true;
+
+        mCardNumberEditText.setOnFocusChangeListener(new OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View v, boolean hasFocus) {
+                if (hasFocus) {
+                    scrollLeft();
+                }
+            }
+        });
+
+        mExpiryDateEditText.setOnFocusChangeListener(new OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View v, boolean hasFocus) {
+                if (hasFocus) {
+                    scrollRight();
+                }
+            }
+        });
+
+        mCvcNumberEditText.setOnFocusChangeListener(new OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View v, boolean hasFocus) {
+                if (hasFocus) {
+                    scrollRight();
+                }
+            }
+        });
+
+        mExpiryDateEditText.setOnKeyListener(new OnKeyListener() {
+            @Override
+            public boolean onKey(View v, int keyCode, KeyEvent event) {
+                if (keyCode == KeyEvent.KEYCODE_DEL
+                        && event.getAction() == KeyEvent.ACTION_DOWN) {
+                    if (mExpiryDateEditText.getText().length() == 0) {
```

Comment:
right -- and i didn't want to swipe back to the left in this case. You might have just made a typo with your first digit and pressed to delete it. I think it would irritate the user to be pushed back into the CC field. As it is, they have to press delete one more time. (delete when there is no text)